### ISSUE TITLE
CustomModelData Pickaxe Mining Bugs

### ISSUE THREAD

**Issue by alexanderdidio** [GeyserMC/Geyser#3345]
### Describe the bug

If you have a pickaxe with CustomModelData (texture pack), even if it's a netherite_pickaxe, it performs as if it's a wooden pickaxe. This is only an issue on Bedrock (Geyser) and not Minecraft: Java Edition.

### To Reproduce

1. Create a netherite_pickaxe with CustomModelData
2. Mine blackstone or obsidian to reproduce

### Expected behaviour

Expect a netherite_pickaxe with custommodeldata to mine with the same default vanilla efficiency.

### Server Version and Plugins

Pufferfish version git-Pufferfish-38 (MC: 1.19.2) (Implementing API version 1.19.2-R0.1-SNAPSHOT)
Geyser version 2.1.0-SNAPSHOT (git-master-c31bd45) (Java: 1.19.1 - 1.19.2, Bedrock: 1.19.0/1.19.2 - 1.19.30)
Velocity 3.1.2-SNAPSHOT (git-3293f7a9-b172)

### Geyser Dump

https://dump.geysermc.org/Umvm1ucqloyp5Ree27N37LkVFrqzikVM

### Geyser Version

Geyser version 2.1.0-SNAPSHOT

### Minecraft: Bedrock Edition Device/Version

Windows, Playstation, Xbox, Nintendo Switch, Mobile

**Comment by YellowZaki** [comment]
Related to https://github.com/GeyserMC/Geyser/issues/3299 and https://github.com/GeyserMC/Geyser/issues/3298

**Comment by Kas-tle** [comment]
> Related to #3299 and #3298

This is not really related to either of these

**Comment by Kas-tle** [comment]
Hello, could you please check if this is still an issue with the latest version of Geyser on a **1.19.40 Bedrock Client**?

**Comment by YellowZaki** [comment]
I've tried with latest Geyser and 1.19.40 bedrock and the issue is still happening. It is not as slow as a wooden pickaxe as OP says, but it is slower than a non textured pickaxe.
In addition, there is no haptics when mining on mobile bedrock client.

**Comment by onebeastchris** [comment]
Duplicate of https://github.com/GeyserMC/Geyser/issues/3348; ill close this one since the other had more activity. A fix for this should be coming once the custom block PR is merged.
